quickwit-index/src/actors/indexer.rs

```diff
@@ -160,6 +160,10 @@ impl ImmutableState {
         ctx: &ActorContext<Indexer>,
     ) -> Result<(), ActorExitStatus> {
         let indexed_split = self.get_or_create_current_indexed_split(current_split_opt, ctx)?;
+        indexed_split
+            .checkpoint_delta
+            .add(batch.checkpoint_delta)
+            .with_context(|| "Batch delta does not follow indexer checkpoint")?;
```

Comment:
It would be better to keep the error details here

Reply:
and add a test for this use case, they may be useful, when we have distributed indexing